src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  static removeDiedAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.name = name;
+  }
+
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
```

Comment:
let's use shortcuts for booleans
```suggestion
    if (animal instanceof Herbivore && !animal.hidden) {
```